test/groovy/MtaBuildTest.groovy

```diff
@@ -278,6 +278,14 @@ public class MtaBuildTest extends BasePiperTest {
         assert shellRule.shell.find(){ c -> c.contains('java -jar /opt/sap/mta/lib/mta.jar --mtar com.mycompany.northwind.mtar --build-target=NEO --extension=config_extension build')}
     }
 
+    @Test
+    void canConfigureMTARName() {
+
+        stepRule.step.mtaBuild(script: nullScript, mtarName: 'custom.name.mtar')
+
+        assert shellRule.shell.find(){ c -> c.contains('--mtar custom.name.mtar')}
```

Comment:
I was also skeptical about that but now, changed the behaviour to treat the config as only name without extension. I have also updated the code doc to make it transparent.

Reply:
Providing the file name without extension looks counter-intuitive for me. OK, it is documented ...
There are a lot of cases where it is clear from the context what the file extension will be, but nevertheless the files are normally provided with file extension. e.g. `tar`, `zip`. They expect the full file name - with appendix - to be provided.

I would like to suggest to change that behaviour. Since it is documented and it will work I don't vote with "require changes", but again, it is very counterintuitive - at least for me.

Reply:
I think if we provide control of the filename to the user, he should also have full control over it. So I also would vote for having the `.mtar` **not** added to the filename automatically.

Reply:
I agree with marcusholl and CCFenner with a single exception when no file extension was provided. Only then, .mtar should be added, so to say as a meaningful default. I'd see this as some added intelligence to the tool, like good human-machine-interfaces do nowadays.

Reply:
so something like `if(mtaName.indexOf('.') == -1) mtaName += '.mtar'`?

Reply:
> when no file extension was provided. Only then, .mtar should be adde

This is not that cool from my personal point of view.  I would prefer to avoid such bells and whistles. Most simple approach is: provide a file name - and that file name is used. No coding inbetween which claims to knows it better. The more bells and whistles the harder to maintain the code. E.g. it is not about one check if the file name contains a dot. We need also test cases for both cases. That overhead is not justified in my personal opinion for having a file appendix "fixed".

Reply:
It's up to you ;)
Like in real world, many people like to hear bells-like sound when they forget to put on the car's safety belt, others don't need it, and some even hate it. And I'm sure the car producer added a fair amount of time for tests of this warning system :)

Reply:
@marcusholl I have changed the new config behaviour to take the custom name with extension.